cpp/perspective/src/cpp/gnode.cpp

```diff
@@ -345,264 +236,150 @@ t_gnode::_process_table() {
                 } break;
                 case DTYPE_UINT8: {
-                    _process_helper<std::uint8_t>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn,
-                        tcolumn, op_base, lkup, prev_pkey_eq_vec, added_offset);
+                    _process_column<std::uint8_t>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn, tcolumn, _process_state);
                 } break;
                 case DTYPE_FLOAT64: {
-                    _process_helper<double>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn,
-                        tcolumn, op_base, lkup, prev_pkey_eq_vec, added_offset);
+                    _process_column<double>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn, tcolumn, _process_state);
                 } break;
                 case DTYPE_FLOAT32: {
-                    _process_helper<float>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn, tcolumn,
-                        op_base, lkup, prev_pkey_eq_vec, added_offset);
+                    _process_column<float>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn, tcolumn, _process_state);
                 } break;
                 case DTYPE_BOOL: {
-                    _process_helper<std::uint8_t>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn,
-                        tcolumn, op_base, lkup, prev_pkey_eq_vec, added_offset);
+                    _process_column<std::uint8_t>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn, tcolumn, _process_state);
                 } break;
                 case DTYPE_TIME: {
-                    _process_helper<std::int64_t>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn,
-                        tcolumn, op_base, lkup, prev_pkey_eq_vec, added_offset);
+                    _process_column<std::int64_t>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn, tcolumn, _process_state);
                 } break;
                 case DTYPE_DATE: {
-                    _process_helper<std::uint32_t>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn,
-                        tcolumn, op_base, lkup, prev_pkey_eq_vec, added_offset);
+                    _process_column<std::uint32_t>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn, tcolumn, _process_state);
                 } break;
                 case DTYPE_STR: {
-                    _process_helper<std::string>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn,
-                        tcolumn, op_base, lkup, prev_pkey_eq_vec, added_offset);
+                    _process_column<std::string>(fcolumn, scolumn, dcolumn, pcolumn, ccolumn, tcolumn, _process_state);
                 } break;
                 default: { PSP_COMPLAIN_AND_ABORT("Unsupported column dtype"); }
             }
         }
 #ifdef PSP_PARALLEL_FOR
     );
 #endif
-    std::shared_ptr<t_data_table> flattened_masked
-        = mask.count() == flattened->size() ? flattened : flattened->clone(mask);
-    PSP_GNODE_VERIFY_TABLE(flattened_masked);
-#ifdef PSP_GNODE_VERIFY
```

Comment:
Here too (I think this is removal not refactor?)

Reply:
I removed the call to verify table but refactored the structure of the `flattened_masked` assignment.